packages/back-end/src/services/features.ts

```diff
@@ -81,6 +82,33 @@ export function getFeatureDefinition({
     rules:
       rules
         ?.filter((r) => r.enabled)
+        ?.filter((r) => {
+          //Filter our rules that have schedulingRules if the current scheduleRule sets the feature to disabled.
+          if (!r.scheduleRules || !r.scheduleRules.length) {
+            return true;
+          }
+
+          const currentDate = new Date().valueOf();
+
+          // Loop through the list of rules, and find the next rule to be applied
+          const nextRuleIndex = r.scheduleRules.findIndex(
+            (rule) =>
+              rule.timestamp !== null &&
+              new Date(rule.timestamp).valueOf() > currentDate
+          );
+
+          if (nextRuleIndex === -1) {
+            if (
+              r.scheduleRules[r.scheduleRules.length - 1].timestamp === null
```

Comment:
nit, for fun: I think we can now get last element using `myarray.at(-1)`

Reply:
Today I learned!